utils/cautils/bootstrap.go

```diff
@@ -0,0 +1,302 @@
+			return errors.Wrap(err, "error storing new default context")
+		}
+		if err := step.Contexts().SetCurrent(context); err != nil {
+			return errors.Wrap(err, "error setting context '%s'")
+		}
+	} else {
+		WarnContext()
+	}
+	rootFile := pki.GetRootCAPath()
+	configFile := step.DefaultsFile()
+
+	if err = os.MkdirAll(filepath.Dir(rootFile), 0700); err != nil {
+		return errs.FileError(err, rootFile)
+	}
+
+	if err = os.MkdirAll(filepath.Dir(configFile), 0700); err != nil {
+		return errs.FileError(err, configFile)
+	}
+
+	// Serialize root
+	_, err = pemutil.Serialize(resp.RootPEM.Certificate, pemutil.ToFile(rootFile, 0600))
+	if err != nil {
+		return err
+	}
+	ui.Printf("The root certificate has been saved in %s.\n", rootFile)
+
+	// make sure to store the url with https
+	caURL, err = utils.CompleteURL(caURL)
+	if err != nil {
+		return err
+	}
+
+	// Serialize defaults.json
+	b, err := json.MarshalIndent(bootstrapConfig{
+		CA:          caURL,
+		Fingerprint: fingerprint,
+		Root:        pki.GetRootCAPath(),
+		Redirect:    bc.redirectURL,
+	}, "", "  ")
+	if err != nil {
+		return errors.Wrap(err, "error marshaling defaults.json")
+	}
+
+	ctx.Set("ca-url", caURL)
+	ctx.Set("fingerprint", fingerprint)
+	ctx.Set("root", rootFile)
+
+	if err := utils.WriteFile(configFile, b, 0644); err != nil {
+		return err
+	}
+
+	ui.Printf("The authority configuration has been saved in %s.\n", configFile)
+
+	if step.Contexts().Enabled() {
+		profileDefaultsFile := step.ProfileDefaultsFile()
+
+		if err := os.MkdirAll(filepath.Dir(profileDefaultsFile), 0700); err != nil {
+			return errs.FileError(err, profileDefaultsFile)
+		}
+
+		if _, err := os.Stat(profileDefaultsFile); os.IsNotExist(err) {
+			if err := ioutil.WriteFile(profileDefaultsFile, []byte("{}"), 0600); err != nil {
+				return errs.FileError(err, profileDefaultsFile)
+			}
+			ui.Printf("The profile configuration has been saved in %s.\n", profileDefaultsFile)
+		}
+	}
+
+	if ctx.Bool("install") {
+		ui.Printf("Installing the root certificate in the system truststore... ")
+		if err := truststore.InstallFile(rootFile); err != nil {
+			ui.Println()
+			return err
+		}
+		ui.Println("done.")
+	}
+
+	return nil
+}
+
+// BootstrapTeamAuthority does a request to api.smallstep.com to bootstrap the
+// configuration of a given team/authority.
+func BootstrapTeamAuthority(ctx *cli.Context, team, teamAuthority string) error {
+	apiEndpoint := ctx.String("team-url")
+	if apiEndpoint == "" {
+		// Use the default endpoint..
+		u := url.URL{
+			Scheme: "https",
+			Host:   "api.smallstep.com",
+			Path:   "/v1/teams/" + team + "/authorities/" + teamAuthority,
+		}
+		apiEndpoint = u.String()
+	} else {
+		// The user specified a custom endpoint..
+		apiEndpoint = strings.ReplaceAll(apiEndpoint, "<>", team)
+		u, err := url.Parse(apiEndpoint)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing %s", apiEndpoint)
+		}
+		apiEndpoint = u.String()
```

Comment:
Comment that this does not support teamAuthority substitution, or add support for it.

Reply:
I added a line to the `--team-url` flag and left a TODO with a comment above the code block.